### PR TITLE
Refactor Polynomials: Extract Method to Introduce RootFinder Factory Method and Tighten Encapsulation

### PR DESCRIPTION
**Pull request by hemalvarambhia**
- Extracted a factory method that generates a root finder with a given precision;
- added some missing tests that felt interesting.
Another test might be to test roots that are in the Complex plane.

### CODE REVIEW

File: src/Math-Tests-Polynomials/PMPolynomialTest.class.st [hunk 1]
```diff
@@ -235,13 +235,36 @@ PMPolynomialTest >> testPolynomialRoots [
...
-	roots := polynomial roots asSortedCollection asArray.
+	roots := polynomial roots .
 	self assert: roots size equals: 3.
 	self assert: (roots at: 1) + 2 closeTo: 0.
 	self assert: (roots at: 2) + 1 closeTo: 0.
 	self assert: (roots at: 3) - 5 closeTo: 0
 ]
 
+{ #category : #'iterative algorithms' }
+PMPolynomialTest >> testPolynomialRootsForConstant [
```
Comment (hemalvarambhia) [inline src/Math-Tests-Polynomials/PMPolynomialTest.class.st:246]:
```suggestion
PMPolynomialTest >> testPolynomialRootsConstantsHaveNoRoots [
```

File: src/Math-Polynomials/PMPolynomial.class.st [hunk 2]
```diff
@@ -26,6 +26,15 @@ PMPolynomial class >> coefficients: anArray [
 	^self new initialize: anArray reverse
 ]
 
+{ #category : #information }
```
Comment (Ducasse) [inline src/Math-Polynomials/PMPolynomial.class.st]: I'm not sure I would put this method on the class side but I could understand. 
Now why the PMNewtonZeroFinder does not define a creation class method  desiredPrecision: like that you do not need this method and the class PMNewtonZeroFinder gets improved.

Comment (Ducasse) [reply src/Math-Polynomials/PMPolynomial.class.st]: I saw that you did it :)

Comment (hemalvarambhia) [reply src/Math-Polynomials/PMPolynomial.class.st]: Yes, I had to take smaller steps in order to keep our tests green.

**Pull request merged**